### ISSUE TITLE
Client Side Work

### ISSUE THREAD

**Issue by susalulmumaO12** [humam154/InternetApplications#7]
ReactJs will be the tech to develop client side.

**Comment by susalulmumaO12** [comment]
Born to be a designer 💅🏼  forced to be not

![image](https://github.com/user-attachments/assets/cca99f22-4bdc-4163-bf51-05a6d6a51f4c)

**Comment by ibra-kdbra** [comment]
GG, fix the layout..

**Comment by ibra-kdbra** [comment]
NOoooooooo, don't use create-react-app, it's dead.
Use vite.js like this:
```shell
# npm 7+, extra double-dash is needed:
npm create vite@latest <this should be your project name> -- --template react
```



> with vite you use .jsx files, not js files, because include functional components in the new versions(18+), it's easier anyway.

**Comment by ibra-kdbra** [comment]
Not me being too meddlesome, but your back-end need work, security logging, tokens response, and some basics in Java I hate.😁

Anyway, why **Spring** over **node**, **laravel**, _this is a small project to make_.

**Comment by susalulmumaO12** [comment]
Hmmm that explains why it sucked, but I'm already using .jsx files, and I'm faking it till I make it (or I don't).

Thanks for the advice.

Yes the backend is weak, why spring boot, because Aspect Oriented Programming, I know it can be applied in Node, but Spring Boot seemed to be more suitable, I don't like it at all. Thank you for the feedback tho.

Wdym tokens response?

**Comment by ibra-kdbra** [comment]
Not me being too meddlesome, but your back-end need work, security logging, tokens response, and some basics in Java I hate.😁

Anyway, why **Spring** over **node**, **laravel**, _this is a small project to make_.

**Comment by ibra-kdbra** [comment]
Nope, node.js maybe can do this, but spring boot is more suitable for AOP.

Check out the time for tokens send from logging out, logging in, and every setup.

Also check [this](https://spring.io/blog/2024/03/22/reflectionless-templates-with-spring), also [this](https://medium.com/@aedemirsen/generic-api-response-with-spring-boot-175434952086)

**Comment by ibra-kdbra** [comment]
> Hmmm that explains why it sucked, but I'm already using .jsx files, and I'm faking it till I make it (or I don't).

Or you will, you can do better.

> Yes the backend is weak.

Who said that, not me, needs work==> fix your code ==> have a coffee watch DB, then sleep.

**Comment by susalulmumaO12** [comment]
> Also check this

Thank you that was valuable, I've been looking for something similar for days, I looked at thymeleaf, I couldn't get it to work.

> ,and this.

This one also seems comprehensive, thank you, really.

> Who said that

I did, from my perspective, I still don't quite get the concept of it, it's overwhelming...

**Comment by ibra-kdbra** [comment]
Can you elaborate more?

Use telegram for faster conversation, I don't like the discussion here.

**Comment by susalulmumaO12** [comment]
I don't get the entire concept of AOP, is it just logging before and after? Then what? What's the exact point? I read many articles about it, it's still not getting into my head.

I think I need to see the bigger picture of it.

**Comment by ibra-kdbra** [comment]
telegram?

**Comment by susalulmumaO12** [comment]
Yes
